### PR TITLE
Update xDS protos

### PR DESCRIPTION
**Pull request by lidizheng**
* envoyproxy/data-plane-api: 9edfeb841b0f8e54816f0b81949f79104072072c
* google/googleapis: 82944da21578a53b74e547774cf62ed31a05b841
* envoyproxy/protoc-gen-validate: 872b28c457822ed9c2a5405da3c33f386ac0e86f
* cncf/udpa: cc1b757b3eddccaaaf0743cbb107742bb7e3ee4f

For xDS reviewers, UDPA moved their core protos from `udpa/core/v1` to `xds/core/v3`. See original pr: https://github.com/cncf/udpa/pull/40.

### CODE REVIEW

**Review by jtattermusch — APPROVED** [review]
granting owner's approval - it would be good for someone who knows more about the XDS related functionality to approve as well.

**Comment by lidizheng** [comment]
Known failures: https://github.com/grpc/grpc/issues/25368 https://github.com/grpc/grpc/issues/22411 https://github.com/grpc/grpc/issues/25369

**Review by markdroth — APPROVED** [review]
Just a couple of minor nits; otherwise, looks great!

File: BUILD [hunk 1]
```diff
@@ -3134,20 +3134,20 @@ grpc_cc_library(
 grpc_cc_library(
     name = "udpa_core_upb",
```
[the remark this answers is not in the document]
Comment (lidizheng) [reply BUILD]: Both Bazel rules are renamed to better reflect its new proto file location.

**Pull request merged**